### PR TITLE
Use only the CrypTensor from Crypten

### PR DESCRIPTION
**Pull request by gmuraru**
Signed-off-by: George Muraru <murarugeorgec@gmail.com>

### CODE REVIEW

File: syft/frameworks/crypten/hook/hook.py [hunk 1]
```diff
@@ -12,10 +13,10 @@ def get_hooked_crypten_func(func_api_name, func):
     @tracer(func_name=cmd_name)
     @wraps(func)
     def overloaded_func(*args, **kwargs):
-        try:
+        if syft.hook.trace.active:
```
Comment (gmuraru) [inline syft/frameworks/crypten/hook/hook.py:16]: Q: I needed to check if the tracer was or not active (tracer is active if we build a plan)

File: syft/frameworks/torch/hook/hook.py [hunk 2]
```diff
@@ -136,11 +135,15 @@ def __init__(
         else:
             self.local_worker.hook = self
 
-        self.to_auto_overload = {}
+        import crypten
```
Comment (gmuraru) [inline syft/frameworks/torch/hook/hook.py:138]: TODO: Those should be moved to the ```hook_crypten method```

File: syft/execution/role.py [hunk 3]
```diff
@@ -15,12 +15,19 @@
...
 from syft.generic.object_storage import ObjectStorage
 from syft.workers.abstract import AbstractWorker
 
 from syft_proto.execution.v1.role_pb2 import Role as RolePB
 
+from syft import dependency_check
+
+if dependency_check.crypten_available:
+    # TODO Crypten shouldn't be used here
+    import crypten
```
Comment (karlhigley) [inline syft/execution/role.py:29]: This part can move into [the (newly created) list of `framework_packages`](https://github.com/OpenMined/PySyft/blob/master/syft/generic/frameworks/types.py#L27), which are available during the execution of `Plans`/`Roles`/`Actions` and used to execute package-level methods.

File: syft/execution/role.py [hunk 4]
```diff
@@ -143,7 +150,7 @@ def build_placeholders(self, obj):
             return type(obj)(r)
         elif isinstance(obj, dict):
             return {key: self.build_placeholders(value) for key, value in obj.items()}
-        elif isinstance(obj, FrameworkTensor):
+        elif isinstance(obj, FrameworkTensor) or isinstance(obj, AbstractTensor):
```
Comment (karlhigley) [inline syft/execution/role.py:153]: I think we should add the Crypten tensor type to `FrameworkTensor`.

Comment (gmuraru) [reply syft/execution/role.py:153]: Moved this part into my PR from here: https://github.com/OpenMined/PySyft/pull/3254 :D

**Comment by gmuraru** [comment]
Will close this since I added those changes in my latest commit from [here](https://github.com/OpenMined/PySyft/pull/3254)

**Pull request closed without merging**